scripts/gas-refund-program/staking/covalent.ts

```diff
@@ -24,15 +25,43 @@ export async function getTokenHolders({
     'page-size': pageSize?.toString(),
   });
 
-  const url = `https://api.covalenthq.com/v1/${chainId}/tokens/${token}/token_holders/?key=${COVALENT_API_KEY}${
+  const url = `/${chainId}/tokens/${token}/token_holders/?key=${COVALENT_API_KEY}${
     queryString ? '&' + queryString : ''
   }`;
 
-  const { data } = await Utils._get<TokensHoldersResponse>(url, 2000); // times out otherwise
+  const { data } = await covalentClient.get<TokensHoldersResponse>(url);
 
   return data.data;
 }
 
+interface TransactionQueryOptions {
+  chainId: number;
+  txHash: string;
+}
+
+interface TransactionResponse {
+  data: {
+    items: [
+      {
+        gas_spent: number;
+      },
+    ];
+  };
+}
+
+export async function getTransactionGasUsed({
+  chainId,
+  txHash,
+}: TransactionQueryOptions): Promise<number> {
+  const url = `/${chainId}/transaction_v2/${txHash}/?key=${COVALENT_API_KEY}`;
+
+  const { data } = await covalentClient.get<TransactionResponse>(url);
+
+  assert(data.data.items.length === 1, 'Expected exactly one transaction');
```

Comment:
So types says what's expected from the api but this is a protection in case something goes wrong. I think I thought relax types before and restrict later ?